src/ghdcbot/engine/orchestrator.py

```
@@ -0,0 +1,436 @@
+            merge_role_rules=merge_role_rules,
+        )
+
+    def close(self) -> None:
+        for adapter in {self.github_reader, self.github_writer, self.discord_reader, self.discord_writer}:
+            close = getattr(adapter, "close", None)
+            if callable(close):
+                close()
+
+
+def _send_notifications_for_new_events(
+    contributions: list[ContributionEvent],
```

Ruff (0.14.14)

[error] Undefined name `ContributionEvent`

(F821)